caf-audit-service-admin/src/main/java/com/hpe/caf/services/audit/server/admin/HealthCheck.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2015-2020 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.services.audit.server.admin;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hpe.caf.auditing.AuditChannel;
+import com.hpe.caf.auditing.AuditConnection;
+import com.hpe.caf.auditing.AuditConnectionFactory;
+import com.hpe.caf.auditing.healthcheck.HealthResult;
+import com.hpe.caf.auditing.healthcheck.HealthStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+
+@WebServlet("/healthcheck")
+public class HealthCheck extends HttpServlet
+{
+    private static final Logger LOG = LoggerFactory.getLogger(HealthCheck.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @Override
+    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp) throws IOException
+    {
+        HealthResult healthResult;
+        try (final AuditConnection connection = AuditConnectionFactory.createConnection()) {
+            try (final AuditChannel channel = connection.createChannel()) {
+                healthResult = channel.healthCheck();
+                if (healthResult.getStatus() != HealthStatus.HEALTHY) {
+                    resp.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Double-check that this is the same code that the workers and other services when their health checks fail.

Reply:
It's what Job-service currently returns, I'm not sure what workers do.